### ISSUE TITLE
process has died

### ISSUE THREAD

**Issue by hsoltani** [AutonomyLab/ardrone_autonomy#115]
I get the following messages when I launch the driver (roslaunch ardrone_autonomy  ardrone.launch):

``` bash
... logging to /home/gcn/.ros/log/9b5be772-074e-11e4-a84f-94dbc99d5609/roslaunch-gcn-K53E-14678.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://gcn-K53E:55264/

SUMMARY
========

CLEAR PARAMETERS
 * /ardrone_driver/

PARAMETERS
 * /ardrone_driver/altitude_max: 3000
 * /ardrone_driver/altitude_min: 50
 * /ardrone_driver/bitrate: 4000
 * /ardrone_driver/control_vz_max: 700
 * /ardrone_driver/control_yaw: 1.75
 * /ardrone_driver/cov/imu_av: [1.0, 0.0, 0.0, 0...
 * /ardrone_driver/cov/imu_la: [0.1, 0.0, 0.0, 0...
 * /ardrone_driver/cov/imu_or: [1.0, 0.0, 0.0, 0...
 * /ardrone_driver/detect_type: 10
 * /ardrone_driver/detections_select_h: 32
 * /ardrone_driver/detections_select_v_hsync: 128
 * /ardrone_driver/do_imu_caliberation: False
 * /ardrone_driver/enemy_colors: 3
 * /ardrone_driver/enemy_without_shell: 0
 * /ardrone_driver/euler_angle_max: 0.21
 * /ardrone_driver/flight_without_shell: 0
 * /ardrone_driver/max_bitrate: 4000
 * /ardrone_driver/navdata_demo: 0
 * /ardrone_driver/outdoor: 1
 * /ardrone_driver/tf_prefix: mydrone
 * /rosdistro: <...>
 * /rosversion: <...>

NODES
  /
    ardrone_driver (ardrone_autonomy/ardrone_driver)

auto-starting new master
process[master]: started with pid [14690]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 9b5be772-074e-11e4-a84f-94dbc99d5609
process[rosout-1]: started with pid [14703]
started core service [/rosout]
process[ardrone_driver-2]: started with pid [14721]
Wait authentification
Wait authentification
Wait authentification
===================+> 192.168.1.1
[ INFO] [1404899494.632161879]: Root Frame is: 0
[ INFO] [1404899494.632911644]: ~cov/imu_la set to [0.1, 0, 0, 0, 0.1, 0, 0, 0, 0.1]
[ INFO] [1404899494.633767884]: ~cov/imu_av set to [1, 0, 0, 0, 1, 0, 0, 0, 1]
[ INFO] [1404899494.634351636]: ~cov/imu_or set to [1, 0, 0, 0, 1, 0, 0, 0, 100000]
Deleting Profile -cea48d22
Deleting Application -e182b69b
Input device Teleop added
Starting thread video_stage
Starting thread video_recorder

video stage thread initialisation

Video multisocket : init 2 sockets
Video multisocket : connecting socket 0 on port 5555 UDP
Video recorder thread initialisation
Starting thread update_ros
Thread `update_ros` started 
 Failed to open AT
Starting thread navdata_update
Starting thread ardrone_control
 - Connection failed
Video multisocket : connecting socket 1 on port 5555 TCP
VP_Com : Failed to open socket for navdata
PA : MEMORY SPACE ALLOWED : 40 MB
Start thread thread_academy_download 
Academy download stage resumed
Start thread thread_academy
Start thread thread_academy_upload
Academy download stage paused
[ardrone_driver-2] process has died [pid 14721, exit code -11, cmd /home/gcn/catkin_ws/devel/lib/ardrone_autonomy/ardrone_driver __name:=ardrone_driver __log:=/home/gcn/.ros/log/9b5be772-074e-11e4-a84f-94dbc99d5609/ardrone_driver-2.log].
log file: /home/gcn/.ros/log/9b5be772-074e-11e4-a84f-94dbc99d5609/ardrone_driver-2*.log
```

Also I get segmenation fault error when I run rosrun ardrone_autonomy ardrone_driver:

``` bash
Wait authentification
Wait authentification
Wait authentification
===================+> 192.168.1.1
[ INFO] [1404899704.002702254]: Root Frame is: 0
[ INFO] [1404899704.003710526]: No values found for `~cov/imu_la` in parameters, set all to zero.
[ INFO] [1404899704.004999033]: No values found for `~cov/imu_av` in parameters, set all to zero.
[ INFO] [1404899704.006706801]: No values found for `~cov/imu_or` in parameters, set all to zero.
Deleting Profile -cea48d22
Deleting Application -e182b69b
Input device Teleop added
Starting thread video_stage
Starting thread video_recorder

video stage thread initialisation

Video multisocket : init 2 sockets
Video recorder thread initialisation
Starting thread update_ros
Video multisocket : connecting socket 0 on port 5555 UDP
Thread `update_ros` started 
 Failed to open AT
Starting thread navdata_update
 - Connection failed
Video multisocket : connecting socket 1 on port 5555 TCP
Starting thread ardrone_control
VP_Com : Failed to open socket for navdata
PA : MEMORY SPACE ALLOWED : 40 MB
Start thread thread_academy_download 
Academy download stage resumed
Start thread thread_academy
Start thread thread_academy_upload
Academy download stage paused
Segmentation fault (core dumped)
```

How can I fix this problem?

**Comment by mani-monaj** [comment]
Could you please provide more information about what operating system you are using and how you have installed the driver. Have you tried any other machine or drone?

**Comment by hsoltani** [comment]
I have tried it on Ubuntu 12.04, 12.10, 13.10 and three versions of ROS (fuerte, hydro & Indigo). But the driver just installed correctly on Ubuntu 12.04 and hydro version and I get these error messages in other situations.

**Comment by kbogert** [comment]
Giving us the backtrace from the segfault could help figure this out, follow the following steps:
-  Clone ardrone_autonomy into your hydro workspace
-  Delete the Build, Devel, and Install folders in your workspace
-  Edit lines 118 and 119 of ardrone_autonomy/ardrone_lib.patch to read:

```
+       @$(MAKE) -C $(SDK_PATH)/Soft/Build $(TMP_SDK_FLAGS) $(SDK_FLAGS) $(MAKECMDGOALS) USE_LINUX=yes BUILD_MODE=DEBUG_MODE
+       @$(MAKE) -C $(SDK_PATH)/VP_SDK/Build $(TMP_SDK_FLAGS) $(SDK_FLAGS) $(MAKECMDGOALS) USE_LINUX=yes BUILD_MODE=DEBUG_MODE
```
-  Compile your workspace with the following command:

```
catkin_make -DCMAKE_BUILD_TYPE=Debug
```
-  Edit line 3 of ardrone_autonomy/launch/ardrone.launch to read:

```
<node name="ardrone_driver" pkg="ardrone_autonomy" type="ardrone_driver" output="screen" clear_params="true" launch-prefix="gdb -ex run --args">
```
-  Now when you type "roslaunch ardrone_autonomy ardrone.launch" the driver will be run using gdb
-  Once the segfault occurs, use the "bt" command in gdb to get a backtrace and post the results here.
-  If you know how to use gdb, include any other information you find about the error

**Comment by heechul** [comment]
I'm using ubuntu 12.04 and have followed the instructions in the following links. 
http://wiki.ros.org/hydro/Installation/Ubuntu
http://wiki.ros.org/tum_ardrone

However, I got similar segfaults when I try to run the driver. The following logs are what I got
- what I typed on terminal 1. 

$ roslaunch ardrone_autonomy ardrone.launch
- what I typed on terminal 2

$ rosrun ardrone_autonomy ardrone_driver
- what I see on terminal 1
  ...
  [ INFO] [1415133531.051362541]: Successfully connected to 'My ARDrone' (AR-Drone 2.0 - Firmware: 2.4.8) - Battery(%): 67
  [ INFO] [1415133531.051407673]: Navdata Publish Settings:
  [ INFO] [1415133531.051438887]:     Legacy Navdata Mode: On
  [ INFO] [1415133531.051462641]:     ROS Loop Rate: 50 Hz
  [ INFO] [1415133531.051485771]:     Realtime Navdata Publish: Off
  [ INFO] [1415133531.051508968]:     Realtime Video Publish: Off
  [ INFO] [1415133531.051533908]:     Drone Navdata Send Speed: 200Hz (navdata_demo=0)
  [ WARN] [1415133539.726467647]: Shutdown request received.
  [ WARN] [1415133539.726520245]: Reason given for shutdown: [new node registered with same name]
  ROS loop terminated ... 
- what I see on terminal 2
  ...
  Deleting Application -e182b69b
  Input device Teleop added
  Starting thread video_stage
  Starting thread video_recorder
  Video recorder thread initialisation
  Starting thread update_ros
  Failed to open AT
  Starting thread navdata_update
  Starting thread ardrone_control
  PA : MEMORY SPACE ALLOWED : 40 MB
  Start thread thread_academy_download 
  Academy download stage resumed
  video stage thread initialisation
  Video multisocket : init 2 sockets
  Video multisocket : connecting socket 0 on port 5555 UDP
  Connection failed
  Video multisocket : connecting socket 1 on port 5555 TCP
  VP_Com : Failed to open socket for navdata
  Start thread thread_academy_upload
  Thread `update_ros` started 
  Start thread thread_academy
  Academy download stage paused
  Segmentation fault (core dumped)

Any ideas to fix the issue?

Thanks,
